merge/merge.go

```diff
@@ -0,0 +1,129 @@
+package merge
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	internaldb "github.com/wtsi-ssg/wrstat/v4/internal/db"
+	"github.com/wtsi-ssg/wrstat/v4/internal/fs"
+)
+
+const (
+	reloadGrace      = 5 * time.Minute
+	SentinelComplete = "combine.complete"
+)
+
+func Merge(sourceDir, destDir string, removeOld bool) error {
+	de, err := fs.FindLatestCombinedOutputOlderThan(sourceDir, SentinelComplete, reloadGrace)
+	if err != nil {
+		return fmt.Errorf("failed to find database files in source dir: %w", err)
+	}
+
+	if err = copyPreservingTimestamp(de, destDir); err != nil {
+		return fmt.Errorf("failed to copy files to new dest: %w", err)
+	}
+
+	if !removeOld {
+		return nil
+	}
+
+	fi, err := os.Lstat(de)
+	if err != nil {
+		return fmt.Errorf("failed to stat latest directory (%s): %w", de, err)
+	}
+
+	if err := removeFromDirWhenOlderThan(sourceDir, fi.ModTime()); err != nil {
+		return fmt.Errorf("failed to remove old source files: %w", err)
+	}
+
+	return nil
+}
+
+func copyPreservingTimestamp(source, dest string) error {
```

Comment:
All the private funcs here also feel like they should be in the internal fs pkg. Just so that there's one place to look for these kinds of utility funcs when wanting to do similar things in the future.